### PR TITLE
[ts sdk, bcs] improves readme and adds to changelog for BCS v0.5

### PR DESCRIPTION
**Pull request by damirka**

### CODE REVIEW

File: sdk/bcs/README.md [hunk 1]
```diff
@@ -20,29 +20,49 @@ At the high level, BCS gives a set of handy abstractions to (de)serialize data.
...
+
 ### Struct
 
 In BCS structs are merely sequences of fields, they contain no type information but the order in
 which fields are defined. It also means that you can use any field names - they won't affect serialization!
-```
-bcs.registerStructType(<TYPE>, {
-    [<FIELD>]: <FIELD_TYPE>,
+```js
+bcs.registerStructType('<TYPE>'', {
```
Comment (ronny-mysten) [inline sdk/bcs/README.md:48]: Are the doubled-up single quotes intentional? Here and L49.

File: sdk/bcs/CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,12 @@
 
 ## 0.4.0
 
+- Adds support for generics;
```
Comment (Jordan-Mysten) [inline sdk/bcs/CHANGELOG.md:11]: We shouldn't really manually modify this, it should be modified through changesets, so I think really we need to keep that in mind when we write changesets.

Comment (damirka) [reply sdk/bcs/CHANGELOG.md:11]: I understand, and my main motivation for this one and for the future is people who depend on this package. As soon as more people get acquainted with the tech, we’ll have to inform about breaking changes and this one is huge.

File: sdk/bcs/README.md (hunk 1 above)
Comment (damirka) [reply sdk/bcs/README.md:48]: Thanks! Not intentional!

File: sdk/bcs/CHANGELOG.md (hunk 2 above)
Comment (damirka) [reply sdk/bcs/CHANGELOG.md:11]: Do you want me to roll back this change? Does it break pnpm? Or we’re good?

Comment (Jordan-Mysten) [reply sdk/bcs/CHANGELOG.md:11]: I think this is fine to keep, but we should try to avoid this in the future

**Pull request closed without merging**